extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -150,32 +153,45 @@ public function getWikiNotifications( $userId, $wikiId, $readSlice = 5, $counton
 	 * @return array
 	 */
 	public function getCounts( $userId ) {
-		global $wgMemc, $wgCityId;
-
-		wfProfileIn(__METHOD__);
-		$wikiList = $this->getWikiList( $userId );
+		return WikiaDataAccess::cache(
+			$this->getCacheKey( $userId ),
+			self::TTL,
+			function() use ( $userId ) {
+				global $wgMemc, $wgCityId;
+
+				$wikiList = $this->getWikiList( $userId );
+
+				// prefetch data
+				$keys = [];
+				$wno = new WallNotificationsOwner;
+				foreach ( $wikiList as $wiki ) {
+					$keys[] = $this->getKey( $userId, $wiki['id'] );
+					$keys[] = $wno->getKey( $wiki['id'], $userId );
+				}
+				$wgMemc->prefetch( $keys );
+
+				$output = [];
+				$total = 0;
+				foreach ( $wikiList as $wiki ) {
+					$wiki['unread'] = $this->getCount( $userId, $wiki['id'], $wiki['id'] == $wgCityId );
+					$total += $wiki['unread'];
+					// show only Wikis with unread notifications
+					// current Wiki is an exception (show always)
+					if ( $wiki['unread'] > 0 || $wiki['id'] == $wgCityId ) {
+						$output[] = $wiki;
+					}
+				}
+				return $output;
+			}
+		);
+	}
 
-		// prefetch data
-		$keys = [];
-		$wno = new WallNotificationsOwner;
-		foreach ( $wikiList as $wiki ) {
-			$keys[] = $this->getKey( $userId, $wiki['id'] );
-			$keys[] = $wno->getKey( $wiki['id'], $userId );
-		}
-		$wgMemc->prefetch( $keys );
+	private function getCacheKey( $userId ) {
+		return wfMemcKey( __CLASS__, 'getCounts', $userId );
+	}
 
-		$output = [];
-		$total = 0;
-		foreach ( $wikiList as $wiki ) {
-			$wiki['unread'] = $this->getCount( $userId, $wiki['id'], $wiki['id'] == $wgCityId );
-			$total += $wiki['unread'];
-			// show only Wikis with unread notifications
-			// current Wiki is an exception (show always)
-			if ( $wiki['unread'] > 0 || $wiki['id'] == $wgCityId )
-				$output[] = $wiki;
-		}
-		wfProfileOut(__METHOD__);
-		return $output;
+	private function purgeCache( $userId ) {
```

Comment:
Similarly to the above comment, maybe this should be ```purgeCountCache``` to make a clear difference between this count cache and the memsync cache that's being used to cache notifications here.